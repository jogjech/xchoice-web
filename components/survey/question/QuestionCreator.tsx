import React, { FunctionComponent } from "react";
import { Form, Input, Button } from "antd";
import ChoiceCreator from "./choice/ChoiceCreator";
import { PlusOutlined } from "@ant-design/icons";

interface Props {}

const QuestionCreator: FunctionComponent<Props> = () => {
  return (
    <Form.List name="questions">
      {(questions, { add: addQuestion, remove: removeQuestion }) => (
        <>
          {questions.map((question) => (
            <div key={question.key}>
              <p className="sub-title">Question {question.key + 1}</p>
              <Form.Item
                {...question}
                name={[question.name, "questionTitle"]}
                fieldKey={[question.fieldKey, "questionTitle"]}
                label="Question Title"
                rules={[{ required: true, message: "Missing question title" }]}
              >
                <Input placeholder="Question Title" maxLength={100} />
              </Form.Item>
              <ChoiceCreator questionField={question}></ChoiceCreator>
            </div>
          ))}
          <Form.Item>
            <Button
              type="dashed"
              onClick={() =>
                addQuestion({
                  choices: [
                    {
                      text: "",
                    },
                  ],
                })
              }
              block
              icon={<PlusOutlined />}
            >
              Add question
            </Button>
          </Form.Item>
        </>
      )}
    </Form.List>
  );
};

export default QuestionCreator;
