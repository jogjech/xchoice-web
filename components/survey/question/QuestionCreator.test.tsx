import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import QuestionCreator from "./QuestionCreator";

describe("Question creator", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<QuestionCreator></QuestionCreator>);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains form list", () => {
    const formList = wrapper.find("FormList");
    expect(formList.prop("name")).toEqual("questions");
  });
});
