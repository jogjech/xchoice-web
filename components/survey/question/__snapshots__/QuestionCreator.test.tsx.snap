// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question creator renders correctly 1`] = `
<FormList
  name="questions"
>
  <Component />
</FormList>
`;
